modular_sand/code/datums/interactions/lewd_definitions.dm

```diff
@@ -260,16 +260,27 @@
 	return TRUE
 
 /mob/living/proc/moan()
-	if(!(prob(get_lust() / get_lust_tolerance() * 65)))
-		return
-	var/moan = rand(1, 7)
-	if(moan == lastmoan)
-		moan--
-	if(!is_muzzled())
-		visible_message(message = span_lewd("<B>\The [src]</B> [pick("moans", "moans in pleasure")]."), ignored_mobs = get_unconsenting())
-	if(is_muzzled())//immursion
-		audible_message(span_lewd("<B>[src]</B> [pick("mimes a pleasured moan","moans in silence")]."))
-	lastmoan = moan
+	if(is_muzzled())
+		visible_message(span_lewd("<B>[src]</B> [pick("mimes a pleasured moan","moans in silence")]."))
+	else
+		visible_message(message = span_lewd("<B>\The [src]</B> [pick("moans", "moans in pleasure")]."), ignored_mobs = get_unconsenting(), omni = TRUE)
+
+		// Get reference of the list we're using based on gender.
+		var/list/moans
+		if (gender == FEMALE)
+			moans = GLOB.lewd_moans_female
+		else
+			moans = GLOB.lewd_moans_male
+
+		// Pick a sound from the list.
+		var/sound = pick(moans)
+
+		// If the sound is repeated, get a new from a list without it.
+		if (lastmoan == sound)
+			sound = pick(LAZYCOPY(moans) - lastmoan)
```

Comment:
Is there a need for the LAZYCOPY()? The subtraction operator doesn't modify the list, and you don't seem to be concerned with the contents being modified either given the earlier code.